doc/contributor/refinements.md

```diff
@@ -66,16 +65,33 @@ method and so the DeclarationContext can be looked up only once and doesn't need
 to be checked after.
 
 ## Method Dispatch
+Example:
+```ruby
+using M1; # R1.ancestors = [R1, A, B, C, Object, Kernel, ...]
+using M2; # R2.ancestors = [R2, D, E, C, Object, Kernel, ...]
+
+C.new.foo # => ?
+```
 
-During method lookup, if a *refined* method is found, the `DeclarationContext`
-(from the frame) is consulted to see if refinements apply to that specific call.
-This *refined* flag enables to only check methods which have refinements for
-refinements and not every module in the ancestor chain. The same happens for
-`super` calls.
+If there are active refinements for the module is found, method lookup recursively iterates by them to find refinement method in `[R, ..Rn]`'s ancestors. To avoid repeated calculations, we stop the search for the ancestors of `R` when `C` is found and proceed to the next active refinement.
```

Comment:
A few typos, suggestion:
```
During method lookup, if active refinements are found in the the `DeclarationContext`
(from the frame), then we check if any of the ancestors has refinements.
For each refined module, method lookup recursively searches for a refinement method in `[R, ...Rn]`'s ancestors. To avoid repeated calculations, and to not search C and above before other refinements, we stop the search for the ancestors of `R` when `C` is found and proceed to the next active refinement.
```